corehq/apps/app_manager/dbaccessors.py

```diff
@@ -108,16 +108,14 @@ def get_app(domain, app_id, wrap_cls=None, latest=False, target=None):
     from .models import Application
     if not app_id:
         raise Http404()
+    try:
+        app = original_app = Application.get_db().get(app_id)
+    except ResourceNotFound:
+        raise Http404()
+
     if latest:
-        try:
-            original_app = Application.get_db().get(app_id)
-        except ResourceNotFound:
-            raise Http404()
         if not domain:
-            try:
-                domain = original_app['domain']
-            except Exception:
-                raise Http404()
```

Comment:
This was a holdover from when this line used to be `get_db().get(app_id)['domain']`